src/components/ColumnHeader/ColumnHeader.tsx

```diff
@@ -0,0 +1,81 @@
+import { useEffect, useState } from 'react';
+import classNames from 'classnames';
+
+export enum SortDirection {
+  NONE = 'NONE',
+  ASC = 'ASC',
+  DESC = 'DESC',
+}
+
+export interface SortEvent {
+  name: string;
+  direction: SortDirection;
+}
+
+interface ColumnHeaderProps {
+  name: string;
+  onSort: (event: SortEvent) => void;
+}
+
+const ColumnHeader = ({ name, onSort }: ColumnHeaderProps) => {
+  const [sortDirection, setSortDirection] = useState<SortDirection>(SortDirection.NONE);
+
+  useEffect(() => {
+    onSort({
+      name: name,
+      direction: sortDirection,
+    });
+  }, [sortDirection]);
+
+  const doSort = () => {
+    if (sortDirection === SortDirection.NONE) {
+      setSortDirection(SortDirection.ASC);
+    } else if (sortDirection === SortDirection.ASC) {
+      setSortDirection(SortDirection.DESC);
+    } else {
+      setSortDirection(SortDirection.NONE);
+    }
+  };
+
+  return (
+    <>
+      <div
+        className="h-3 grid grid-flow-col-dense w-8 mt-1 ml-1 text-sm text-greyText cursor-pointer hover:text-controlGreyHover"
+        onClick={doSort}
+      >
+        <div>{name.toUpperCase()}</div>
+
+        <div className="mt-1 ml-2.5">
+          <svg
+            className={classNames('hover:stroke-controlGreyHover', {
+              'stroke-controlGreyHover': sortDirection === SortDirection.ASC,
+              'stroke-controlGrey': sortDirection === SortDirection.NONE || sortDirection === SortDirection.DESC,
+            })}
+            width="10"
+            height="6"
+            viewBox="0 0 10 6"
+            fill="none"
+            xmlns="http://www.w3.org/2000/svg"
+          >
+            <path d="M1 5L5 1L9 5" strokeLinecap="round" strokeLinejoin="round" />
+          </svg>
+          <svg
+            className={classNames('hover:stroke-controlGreyHover', {
+              'stroke-controlGreyHover': sortDirection === SortDirection.DESC,
+              'stroke-controlGrey': sortDirection === SortDirection.NONE || sortDirection === SortDirection.ASC,
+            })}
+            width="10"
+            height="6"
+            viewBox="0 0 10 6"
+            fill="none"
+            xmlns="http://www.w3.org/2000/svg"
+          >
+            <path d="M9 1L5 5L1 1" strokeLinecap="round" strokeLinejoin="round" />
+          </svg>
+        </div>
+      </div>
```

Comment:
I've experienced a strange feeling when testing the component so I started to investigate and found a few things with it.

Not 100% relevant to the code comment, but I think it'll be worth to mention so in the future we'd try to test different things - here're few things I noticed during the test:
1. Cursor pointer is lost when moving in between the label and the arrows
1. Arrows have hover effect, but no different action than clicking in the label. The hover effect made me think I could set a particular order instead of alternating, but this it wasn't the case.
1. Arrows are too close
1. If I started to alternate the status relatively quick, the label will be selected (as any other text would), but it felt kind of wrong, because it's more a button rather than text.

Anyway, with these test result, my first step was to turn on the outlines in Storybook. Here's how it looked:
![Storybook ColumnHeader component outline lines](https://user-images.githubusercontent.com/52673485/225930362-6bc20ef1-884d-42c1-8c52-fc904e8998c4.png)

As you can see the outlines shows that there's an issue with the component size. 

After looking at the code, I think we'd do some changes to improve the items mentioned above. Here's some code which possibly will possible fix these issues (with comments explaining why)

```jsx
<div
  // 1. Changed classnames to fix cursor pointer issue & fix sizes shown in the Storybook outline
  // 1.1. Removed height & width as there's no need to set them. This should be automatically set.
  // 1.2. Changed grid system to use flexbox instead. As mentioned in the other comment, flexbox should be used when the layout is for one dimension.
  // 1.3. Added transition to hover state to make it smoother.
  // 1.4. Added items-center to align the text & arrows vertically.
  // 1.5. Removed mt-1 & ml-1 as this should be handled by the parent component.
  className="inline-flex text-sm text-greyText cursor-pointer transition hover:text-controlGreyHover items-center"
  onClick={doSort}
>
  {/*
    4. Changed classnames to make text unselectable
    4.1. Changed `div` to `span` as this is a text element.
    4.2. Added `select-none` to make the text unselectable.
    4.3. Removed .toUpperCase() and added `uppercase` to make the text uppercase. (less computationally expensive)
  */}
  <span className="select-none uppercase">{label}</span>

  {/*
    3. Changed classnames to improve distance between arrows
    3.1. Removed `mt-1` as this wasn't properly aligning vertically. And this is now done (see 1.4. above)
    3.2. Added `space-y-1` to add some vertical distance between the arrows.
  */}
  <div className="ml-2.5 space-y-1">
    {/*
      2. Changed classnames to remove hover state as this was causing the impression that the arrows were doing a different action
      2.1. Removed `hover:stroke-controlGreyHover`.
    */}
    <svg
      className={classNames({
        'stroke-controlGreyHover': sortDirection === SortDirection.ASC,
        'stroke-controlGrey': sortDirection === SortDirection.NONE || sortDirection === SortDirection.DESC,
      })}
      width="10"
      height="6"
      viewBox="0 0 10 6"
      fill="none"
      xmlns="http://www.w3.org/2000/svg"
    >
      <path d="M1 5L5 1L9 5" strokeLinecap="round" strokeLinejoin="round" />
    </svg>
    {/*
      2. Changed classnames to remove hover state as this was causing the impression that the arrows were doing a different action
      2.1. Removed `hover:stroke-controlGreyHover`.
    */}
    <svg
      className={classNames({
        'stroke-controlGreyHover': sortDirection === SortDirection.DESC,
        'stroke-controlGrey': sortDirection === SortDirection.NONE || sortDirection === SortDirection.ASC,
      })}
      width="10"
      height="6"
      viewBox="0 0 10 6"
      fill="none"
      xmlns="http://www.w3.org/2000/svg"
    >
      <path d="M9 1L5 5L1 1" strokeLinecap="round" strokeLinejoin="round" />
    </svg>
  </div>
</div>
```

Reply:
Thanks @Publicker really appreciate You taking the time to look at this.